src/Solver.cpp

```diff
@@ -361,40 +337,18 @@ void Solver::setTrackGenerator(TrackGenerator* track_generator) {
                "since the TrackGenerator has not yet generated tracks");
 
   _track_generator = track_generator;
-  _num_azim = _track_generator->getNumAzim() / 2;
+  _num_polar = _track_generator->getQuadrature()->getNumPolarAngles() / 2;
   _num_parallel_track_groups = _track_generator->getNumParallelTrackGroups();
   _tot_num_tracks = _track_generator->getNumTracks();
   _tracks = _track_generator->getTracksByParallelGroup();
+  _quad = track_generator->getQuadrature();
```

Comment:
perhaps retrieve the `Quadrature` from the `TrackGenerator` into the `_quad` attribute first, then retrieve `_num_polar`? Would have more of a logical flow to it.